pkg/apis/agones/v1/gameserver.go

```diff
@@ -356,6 +394,25 @@ func (gss *GameServerSpec) applySchedulingDefaults() {
 	}
 }
 
+func (gss *GameServerSpec) applyEvictionDefaults() {
+	if !runtime.FeatureEnabled(runtime.FeatureSafeToEvict) {
+		return
+	}
+	if gss.Eviction.Safe == "" {
+		gss.Eviction.Safe = EvictionSafeNever
+	}
+}
+
+func (gs *GameServer) applyEvictionStatus() {
+	if !runtime.FeatureEnabled(runtime.FeatureSafeToEvict) {
+		return
+	}
+	gs.Status.Eviction = gs.Spec.Eviction
+	if gs.Spec.Template.ObjectMeta.Annotations[PodSafeToEvictAnnotation] == "true" {
```

Comment:
Is this the reason why we have a status value?

This feels a little magical to me. I'm not sure it's necessary. I'd be tempted to get rid of gs.status.eviction and just rely on what the user specifies in gs.spec.eviction.

Reply:
This is why there's a status, yes, to make explicit that we are inferring the .Eviction settings rather than using the Spec. The intention was to demystify this particular magic - we only base decisions on the Status. The inference from the annotation was mentioned in the design as necessary for backwards compatibility.

Reply:
Got it. re-read the design notes - that makes sense. Objection rescinded  👍🏻